src/components/AdvancedSearchBar/AdvancedSearchBar.vue

```diff
@@ -0,0 +1,181 @@
+        <div class="search-body m-2 border-b">
+          <div v-for="itemGroup in searchResults" :key="itemGroup">
+            <LobTable v-if="itemGroup" class="min-w-full mb-4" space="sm">
+              <TableHeader>
+                <slot name="header" :result="itemGroup" />
+              </TableHeader>
+              <TableBody>
+                <TableRow
+                  v-for="item in itemGroup?.items"
+                  :key="item"
+                  class="text-gray-500 hover:text-primary-700 cursor-pointer"
+                  @click="hide"
+                >
+                  <slot name="body" :result="item" />
+                </TableRow>
+              </TableBody>
+            </LobTable>
+          </div>
+        </div>
+        <!-- If search has results and user wants to show a footer, the total number of results -->
+        <div
+          v-if="searchTerm && footer"
+          class="search-footer m-2 flow-root text-sm text-gray-700"
+        >
+          <div class="float-left">
+            {{ totalResults }} {{ t('search.matchingResults') }}
+          </div>
+          <div class="float-right flex">
+            <LobLink
+              :to="link"
+              :underline="false"
+              class="text-sm hover:text-primary-900"
+              @click="hide"
+            >
+              {{ t('search.seeAllResults') }}
+              <Icon class="inline" :icon="IconName.NEXT" />
+            </LobLink>
+          </div>
+        </div>
+      </template>
+      <!-- If no results are found show a no results message -->
+      <template v-else>
+        {{ t('search.noResults') }}
+      </template>
+    </div>
+  </div>
+</template>
+<script setup lang="ts">
+import TextInput from '../TextInput/TextInput';
+import LobTable from '../Table/Table';
+import TableHeader from '../Table/TableHeader';
+import TableBody from '../Table/TableBody';
+import TableRow from '../Table/TableRow';
+import LobLink from '../Link/Link.vue';
+import MagnifyingGlass from '../Icons/MagnifyingGlass';
+import XmarkLarge from '../Icons/XmarkLarge';
+import { ref, watch, computed, onMounted, onUnmounted } from 'vue';
+import { Icon, IconName } from '../Icon';
+
+const searchTerm = ref('');
+const searchResults = ref([]);
+const searching = ref(false);
+const visible = ref(false);
+const timeout = ref(null);
+const searchBar = ref(null);
+
+const props = withDefaults(
+  defineProps<{
+    searchFunction: Function;
+    count?: number;
+    link?: string;
+    footer?: boolean;
+  }>(),
+  {
+    count: 0,
+    link: '',
+    footer: true
+  }
+);
+
+const disabled = computed(() => !searchTerm.value);
+const totalResults = computed(() => props.count);
+
+function search(searchTerm: string) {
+  searchResults.value = ref([]);
+  props
+    .searchFunction(searchTerm)
+    .then((results) => {
+      searchResults.value = results;
+    })
+    .finally(() => {
+      searching.value = false;
+    });
```

Comment:
async/ await is easier to read and generally preferred when possible

```suggestion
  searching.value = true; // Not sure if this is needed, I don't see where it gets set to `true`
  searchResults.value = await props.searchFunction(searchTerm.value);
  searching.value = false;
```

Reply:
I don't think the searching.value is needed as the caller is setting it already. Because these are components we don't want to use await as that could have unintended consequences. Hence using a .then() which resolves after the result is available.

Reply:
Actually I just looked at it and because the searchFunction is being called in a setTimeout method it's ok to add await. But generally component libraries functions being async is a little different, because then the consumer would need to know its async which 90% of our components are not.